rust/connlib/tunnel/src/lib.rs

```diff
@@ -224,14 +197,10 @@ pub struct TunnelStats {
     dns_resources: HashMap<String, ResourceDescription>,
     network_resources: HashMap<IpNetwork, ResourceDescription>,
     gateway_public_keys: HashMap<GatewayId, String>,
-
-    awaiting_connection: HashMap<ConnId, AwaitingConnectionDetails>,
-    gateway_awaiting_connection: HashMap<GatewayId, Vec<IpNetwork>>,
```

Comment:
Do we want these to be part of `TunnelStats`? Or should we print all of `TRoleState` instead?

Reply:
I was thinking of removing `TunnelStats` completely in favor of implementing `Debug` instead.

I created `TunnelStats` because there are some costly operations in creating it but we can document that in the `debug` implementation.

In that case we just add `TRoleState` as part of the debug print.

what do you think?

Reply:
It depends a bit on what we want out of the stats. At the moment, it is printed on an interval. I've not worked with connlib much so perhaps this is what we need.

Intuitively, I would have not gone down that route and instead just add logs as things happen and be selective in what we want to print (also in terms of what goes in a span and what do we log as an event).

Reply:
Printing these stats been super useful to debug, finding deadlocks, or resources not being cleaned up and problems with connections.

In the future, though, I want to have some kind of interface(through a socket, dbus or a server) to print the stats on demand.

Reply:
Possibly relevant: #2060